### PR TITLE
[Feature] Support custom sampler in SimpleDatasetStream

### PR DESCRIPTION
**Pull request by eric-haibin-lin**
## Description ##
Add support for custom sampler in SimpleDatasetStream, so that SplitSampler can be passed as an argument. 

### Essentials ###
- [x] PR's title starts with a category (e.g. [BUGFIX], [MODEL], [TUTORIAL], [FEATURE], [DOC], etc)
- [x] Changes are complete (i.e. I finished coding on this PR)
- [x] All changes have test coverage
- [x] Code is well-documented

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

**Comment by mli** [comment]
Job PR-507/1 is complete. 
Docs are uploaded to http://gluon-nlp-staging.s3-accelerate.dualstack.amazonaws.com/PR-507/1/index.html

File: src/gluonnlp/data/stream.py [hunk 1]
```diff
@@ -170,8 +170,8 @@ class SimpleDatasetStream(DatasetStream):
         passed to this class.
     file_pattern: str
         Path to the input text files.
-    file_sampler : str, {'sequential', 'random'}, defaults to 'random'
-        The sampler used to sample a file.
+    file_sampler : str or gluon.data.Sampler, defaults to 'random'
```
Comment (szha) [inline src/gluonnlp/data/stream.py]: I wonder if "file" is the right unit of the chunk of data. Each chunk in a dataset stream should be a dataset, so the API design should reflect that.

Comment (eric-haibin-lin) [reply src/gluonnlp/data/stream.py]: I'm not sure what is the best way to define a "chunk". Currently Corpus/Textline dataset both take "file" as an argument to construct a dataset.

**Comment by eric-haibin-lin** [comment]
I'll merge the sampler change for now. If we need to refine the definition for "chunk", we shall do it in a separate PR

**Pull request merged**